src/psyclone/psyir/backend/fortran.py

```diff
@@ -305,28 +305,16 @@ class FortranWriter(LanguageWriter):
     currently PSyIR algorithm code which has its own gen method for
     generating Fortran).
 
-    :param bool skip_nodes: If skip_nodes is False then an exception \
-        is raised if a visitor method for a PSyIR node has not been \
-        implemented, otherwise the visitor silently continues. This is an \
-        optional argument which defaults to False.
-    :param str indent_string: Specifies what to use for indentation. This \
-        is an optional argument that defaults to two spaces.
-    :param int initial_indent_depth: Specifies how much indentation to \
-        start with. This is an optional argument that defaults to 0.
-    :param bool check_global_constraints: whether or not to validate all \
-        global constraints when walking the tree. Defaults to True.
+    :param kwargs: additional keyword arguments provided to the super class.
+    :type kwargs: unwrapped dict.
 
     '''
     _COMMENT_PREFIX = "! "
 
-    def __init__(self, skip_nodes=False, indent_string="  ",
-                 initial_indent_depth=0, check_global_constraints=True):
+    def __init__(self, **kwargs):
```

Comment:
Nice simplification.

Reply:
Might it be better to leave the comments in place? If I need to find out what parameters I can specify, I first need to find LanguageWriter, which tells me that I need to look at Visitor. I admit that this is not an idea solution either, since then we have to keep the comments across three classes in sync :( Or maybe just add a link to where the valid names are specified?

Reply:
We had the same problem on the PSyIR nodes and Symbols kwargs, and the Transformations options which are passed to the super class. I don't like having them duplicated as we never check them when they are updated, or when new arguments added to the subclasses, and would be quite hard to review.

Adding a link to the super() is a good and simple solution (which some IDEs, static type tools may be able to follow). So I will use this here.

There is a more complex solution of programmatically modifying the `__doc__` to include the BASE.__doc__ or even spit __doc__ to separate the kwargs so that they can be included in the subclass docsting, but this may be over-engineering it, and probably not for this PR.

Reply:
I promoted this comment into an issue and will be looked at independently of this PR